src/main/java/com/myThreadPool/RunnableDelayPairing.java

```diff
@@ -0,0 +1,22 @@
+package com.myThreadPool;
+
+import lombok.Getter;
+
+/**
+ * This is a wrapper class to contain both the Runnable and its delay, after which it is supposed to execute
+ */
+@Getter
+public class RunnableDelayPairing {
+    private final Runnable task;
+    private final long delay;
+
+    public RunnableDelayPairing(Runnable runnable, long delay) {
+        this.task = runnable;
+        this.delay = delay;
+    }
```

Comment:
You can even remove the `RunnableDelayPairing(Runnable runnable, long delay)` constructor and replace it with `@RequiredArgsConstructor` annotation on the class. This annotation instructs lombok to generate a constructor which will take all `final` fields as input arguments.